settings/ui/components/SettingsApp.vue

```diff
@@ -118,21 +122,30 @@ export default {
     getExtensionIcon (extension) {
       extension = extension || ''
       switch (extension) {
-        case 'ocis-accounts': return 'account_circle'
-        case 'ocis-hello': return 'tag_faces'
-        default: return 'application'
+        case 'ocis-accounts':
+          return 'account_circle'
+        case 'ocis-hello':
+          return 'tag_faces'
+        default:
+          return 'application'
       }
     }
   },
-  async created () {
-    await this.initialize()
-    this.resetMenuItems()
-    this.resetSelectedExtension()
+  created () {
+    this.initialize()
   },
   watch: {
-    initialized () {
-      this.resetMenuItems()
-      this.resetSelectedExtension()
+    '$language.current': {
```

Comment:
I fear that this is needed for now. But it looks wrong that extensions have to watch on the language themselves. Only oC web core should be responsible for triggering translation updates on language changes. 🤔

Reply:
yea, no separation of concerns. I pick thist up for our improvement paper